File: test_e2e_job.py

import requests
import time
import json

base_url = 'http://localhost:4000/api/catalog-enricher'
payload = {
    "pricelistId": "652d959-7acd-401a-8c6e-78a84bd875f8",
    "skuColumn": "PrintConfigCode",
    "profileId": "a56b5943-953d-4879-9936-719cdf35ad29"
}

print("🚀 Starting E2E Test: Targeted Enrichment (Python)...")

try:
    # 1. Start Job
    print("Step 1: POST /merger/targeted-enrichment")
    r = requests.post(f"{base_url}/merger/targeted-enrichment", json=payload)
    print(f"Status Code: {r.status_code}")
    print(f"Response: {r.text}")
    
    data = r.json()
    job_id = data.get('jobId')
    
    if not job_id:
        print("❌ FAILURE: No jobId returned!")
        exit(1)

    # 2. Poll Active Jobs
    print(f"\nStep 2: GET /crawler/active-jobs (Poll 1)")
    r_poll = requests.get(f"{base_url}/crawler/active-jobs")
    print(f"Active Jobs count: {len(r_poll.json().get('jobs', []))}")
    print(f"Active Jobs: {json.dumps(r_poll.json().get('jobs', []), indent=2)}")

    jobs = r_poll.json().get('jobs', [])
    job_found = any(j['id'] == job_id for j in jobs)

    if job_found:
        print("✅ SUCCESS: Job found in active list!")
    else:
        print("❌ FAILURE: Job NOT found in active list immediately after creation.")
        
        # Wait 2 seconds and retry
        print("\nWaiting 2s for retry...")
        time.sleep(2)
        
        print("Step 3: GET /crawler/active-jobs (Poll 2)")
        r_poll2 = requests.get(f"{base_url}/crawler/active-jobs")
        jobs2 = r_poll2.json().get('jobs', [])
        job_found2 = any(j['id'] == job_id for j in jobs2)
        
        if job_found2:
            print("✅ SUCCESS: Job found in active list on retry!")
        else:
            print("❌ CRITICAL: Job still missing after 2 seconds.")

except Exception as e:
    print(f"Test Error: {e}")

File: test_e2e_job.js

const axios = require('axios');

async function testTargetedEnrichment() {
    const baseUrl = 'http://localhost:4000/api/catalog-enricher';
    const payload = {
        pricelistId: "652d959-7acd-401a-8c6e-78a84bd875f8",
        skuColumn: "PrintConfigCode",
        profileId: "a56b5943-953d-4879-9936-719cdf35ad29"
    };

    console.log("🚀 Starting E2E Test: Targeted Enrichment...");
    try {
        // 1. Start Job
        console.log("Step 1: POST /merger/targeted-enrichment");
        const startRes = await axios.post(`${baseUrl}/merger/targeted-enrichment`, payload);
        console.log("Response:", startRes.data);
        const jobId = startRes.data.jobId;

        if (!jobId) throw new Error("No jobId returned!");

        // 2. Poll Active Jobs
        console.log("\nStep 2: GET /crawler/active-jobs (Poll 1)");
        const pollRes1 = await axios.get(`${baseUrl}/crawler/active-jobs`);
        console.log("Active Jobs:", JSON.stringify(pollRes1.data.jobs, null, 2));

        const jobFound = pollRes1.data.jobs.find(j => j.id === jobId);
        if (jobFound) {
            console.log("✅ SUCCESS: Job found in active list!");
            console.log("Status:", jobFound.status, "Progress:", jobFound.progress);
        } else {
            console.log("❌ FAILURE: Job NOT found in active list immediately after creation.");

            // Wait 2 seconds and retry
            console.log("\nWaiting 2s for retry...");
            await new Promise(r => setTimeout(r, 2000));

            console.log("Step 3: GET /crawler/active-jobs (Poll 2)");
            const pollRes2 = await axios.get(`${baseUrl}/crawler/active-jobs`);
            console.log("Active Jobs:", JSON.stringify(pollRes2.data.jobs, null, 2));

            if (pollRes2.data.jobs.find(j => j.id === jobId)) {
                console.log("✅ SUCCESS: Job found in active list on retry!");
            } else {
                console.log("❌ CRITICAL: Job still missing after 2 seconds.");
            }
        }

    } catch (e) {
        console.error("Test Failed:", e.response ? e.response.data : e.message);
    }
}

testTargetedEnrichment();
